pandas/tseries/offsets.py

```diff
@@ -427,6 +424,30 @@ def _offset_str(self):
     def nanos(self):
         raise ValueError("{name} is a non-fixed frequency".format(name=self))
 
+    def __setstate__(self, state):
+        """Reconstruct an instance from a pickled state"""
+        if 'offset' in state:
+            # Older versions have offset attribute instead of _offset
+            if '_offset' in state:  # pragma: no cover
+                raise ValueError('Unexpected key `_offset`')
+            state['_offset'] = state.pop('offset')
+            state['kwds']['offset'] = state['_offset']
+
+        if '_offset' in state and not isinstance(state['_offset'], timedelta):
+            # relativedelta, we need to populate using its kwds
+            offset = state['_offset']
+            odict = offset.__dict__
+            kwds = {key: odict[key] for key in odict if odict[key]}
+            state.update(kwds)
```

Comment:
why not just ``state.update(odict)``?

Reply:
Because we don't want to pull in all of the unused relativedelta attributes.